File: DXDAL/MachineLine.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Xpo.Helpers
Imports DevExpress.Data.Filtering

Namespace SPGData
    <Persistent("tblMachineLine")> _
    Public Class MachineLine
        Inherits XPLiteObject
        Dim fMachineLineID As Integer
        <Key(True)> _
        Public Property MachineLineID() As Integer
            Get
                Return fMachineLineID
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue(Of Integer)("MachineLineID", fMachineLineID, value)
            End Set
        End Property
        Dim fMachineLineName As String
        <Size(50)> _
        Public Property MachineLineName() As String
            Get
                Return fMachineLineName
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("MachineLineName", fMachineLineName, value)
            End Set
        End Property

        <Association("Machine-Items", GetType(ItemMachineStandards)), Aggregated()>
        Public ReadOnly Property MachineItems As XPCollection(Of ItemMachineStandards)
            Get
                Return GetCollection(Of ItemMachineStandards)("MachineItems")
            End Get
        End Property
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Sub New()
            MyBase.New(Session.DefaultSession)
        End Sub
        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub
#Region "XPO nested fields class - don't edit manually"
        Public Shadows Class FieldsClass
            Inherits DevExpress.Xpo.PersistentBase.FieldsClass
            Public Sub New()
                MyBase.New()
            End Sub
            Public Sub New(ByVal propertyName As String)
                MyBase.New(propertyName)
            End Sub
            Public ReadOnly Property fMachineLineID() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("fMachineLineID"))
                End Get
            End Property
            Public ReadOnly Property MachineLineID() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("MachineLineID"))
                End Get
            End Property
            Public ReadOnly Property fMachineLineName() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("fMachineLineName"))
                End Get
            End Property
            Public ReadOnly Property MachineLineName() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("MachineLineName"))
                End Get
            End Property
        End Class
        Private Shared _fields As FieldsClass
        Public Shared Shadows ReadOnly Property Fields() As FieldsClass
            Get
                If ReferenceEquals(_fields, Nothing) Then
                    _fields = New FieldsClass()
                End If
                Return _fields
            End Get
        End Property
#End Region
    End Class
End Namespace

File: DXDAL/MachineLine.cs
using DevExpress.Xpo;
using DevExpress.Data.Filtering;

namespace DXDAL
{
    namespace SPGData
    {
        [Persistent("tblMachineLine")]
        public class MachineLine : XPLiteObject
        {
            private int fMachineLineID;
            [Key(true)]
            public int MachineLineID
            {
                get => fMachineLineID;
                set => SetPropertyValue<int>("MachineLineID", ref fMachineLineID, value);
            }
            private string fMachineLineName;
            [Size(50)]
            public string MachineLineName
            {
                get => fMachineLineName;
                set => SetPropertyValue("MachineLineName", ref fMachineLineName, value);
            }

            [Association("Machine-Items", typeof(ItemMachineStandards))]
            [Aggregated()]
            public XPCollection<ItemMachineStandards> MachineItems => GetCollection<ItemMachineStandards>("MachineItems");
            public MachineLine(Session session) : base(session)
            {
            }
            public MachineLine() : base(Session.DefaultSession)
            {
            }
            public override void AfterConstruction()
            {
                base.AfterConstruction();
            }
            public new class FieldsClass : PersistentBase.FieldsClass
            {
                public FieldsClass() : base()
                {
                }
                public FieldsClass(string propertyName) : base(propertyName)
                {
                }
                public OperandProperty fMachineLineID => new OperandProperty(GetNestedName("fMachineLineID"));
                public OperandProperty MachineLineID => new OperandProperty(GetNestedName("MachineLineID"));
                public OperandProperty fMachineLineName => new OperandProperty(GetNestedName("fMachineLineName"));
                public OperandProperty MachineLineName => new OperandProperty(GetNestedName("MachineLineName"));
            }
            private static FieldsClass _fields;
            public static new FieldsClass Fields
            {
                get
                {
                    if (ReferenceEquals(_fields, null))
                        _fields = new FieldsClass();
                    return _fields;
                }
            }
        }
    }
}
